src/mlpack/methods/ann/layer/flatten_t_swish_impl.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file methods/ann/layer/flatten_t_swish_impl.hpp
+ * @author Fawwaz Mayda
+ *
+ * Definition of Flatten T Swish layer first introduced in the acoustic model,
+ * Hock Hung Chieng, Noorhaniza Wahid, Pauline Ong, Sai Raj Kishore Perla,
+ * "Flatten-T Swish: a thresholded ReLU-Swish-like activation function for deep learning", 2018
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_FLATTEN_T_SWISH_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_FLATTEN_T_SWISH_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "flatten_t_swish.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+FlattenTSwish<InputDataType, OutputDataType>::FlattenTSwish(
+    const double T) : tParam(T)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void FlattenTSwish<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  output.set_size(input.n_elem);
+    for(size_t i = 0; i < input.n_elem; ++i) 
+    {
+        if (input.at(i)>=0) 
+        {
+          output.at(i) = (input.at(i) / (1 + std::exp(-input.at(i)))) + tParam;
```

Comment:
If you iterate over all elements you can directly use `input(i)` instead of `input.at(i)`, also I would call `tParam` just `T` just like it's done in the paper.